test/e2e/kubectl/kubectl.go

```diff
@@ -102,6 +102,7 @@ const (
 	httpdDeployment2Filename  = "httpd-deployment2.yaml.in"
 	httpdDeployment3Filename  = "httpd-deployment3.yaml.in"
 	metaPattern               = `"kind":"%s","apiVersion":"%s/%s","metadata":{"name":"%s"}`
+	unknownMetaPattern        = `"kind":"%s","apiVersion":"%s/%s","metadata":{"unknownMeta": "foo", "name":"%s"}`
```

Comment:
I know this is a pre-existing pattern, but it makes for almost unreadable tests ... seems like a generator function like `func unknownFieldMetadataJSON(gvk schema.GroupVersionKind, name string) string { ... }` would be more readable